### ISSUE TITLE
Single-Speaker Training Models CAN NOT Copy another Voice Characteristics

### ISSUE THREAD

**Issue by wanfnc** [SWivid/F5-TTS#686]
### Checks

- [X] This template is only for usage issues encountered.
- [X] I have thoroughly reviewed the project documentation but couldn't find information to solve my problem.
- [X] I have searched for existing issues, including closed ones, and couldn't find a solution.
- [X] I confirm that I am using English to submit this report in order to facilitate communication.

### Environment Details

window 10, Python 3.12.7 & 3.10, NVIDIA GeForce RTX 3080 with 10GB of VRAM

### Steps to Reproduce

Hi guys,

First of all, thank you for your amazing work. I’ve been trying to train a Korean TTS model using F5-TTS (multiple times).

Initially, I attempted to train it with a multi-speaker dataset. While there was almost no quality difference between the reference and generated audio during training, the actual inference only managed to copy the voice timbre and failed to read Korean text correctly.

Later, I came across @lpscr 's advice in discussion #143  and tried training again with a single-speaker dataset of about 12 hours. The training settings were as follows:

Batch size per GPU: 1600
Epochs: 1000
Number of warmup updates: 10000
Save per updates: 25000
Last per steps: 5000
Fine-tune: true
Tokenizer type: char
Mixed precision: bf16
Learning rate: 0.00001

For reference, my GPU is an NVIDIA GeForce RTX 3080 with 10GB of VRAM.

Below is the training log. Although the training didn’t complete all the set epochs due to my computer overheating, it reached approximately 300K steps.

![241230_finallog](https://github.com/user-attachments/assets/f8abbb68-bdcd-422c-b770-8ca58cb7830e)

The inference results from the trained model are as follows:

1) When the voice used in training is provided as the audio prompt, the model produces decent results.
2) When another voice is provided as the audio prompt, it fails to mimic the voice characteristics and only shows very subtle changes in the voice trained on. The text conversion also becomes incomplete.

In conclusion, I have a few questions for you:

1) When training a model with a single speaker, is it inherently limited or impossible to mimic other voices provided as audio prompts?
2) If so, what would be the minimum number or composition of speakers needed to build a model capable of mimicking diverse voices?
3) Is it possible to fine-tune the current model further using a dataset of another speaker’s voice?

Thank you!

**Comment by SWivid** [comment]
Hi, @wanfnc 
1. yes
2. a dozen is minimal i thought, more is better
3. yes, but current model learned only english and chinese yet

> Initially, I attempted to train it with a multi-speaker dataset. While there was almost no quality difference between the reference and generated audio during training, the actual inference only managed to copy the voice timbre and failed to read Korean text correctly.

timbre learned faster, alignment (say pronunciation) harder but will surely be learned

**Comment by wanfnc** [comment]
@SWivid 
Huge Thanks for your reply!

3. the additional training I mentioned was about fine-tuning my own Korean model. 

I have a couple of additional questions:

1) For the speakers I want to add, do I still need at least 10 hours of data per speaker, or can I use much less since the model has already been trained?

2) Is it possible to fine-tune multiple speakers at once? Alternatively, can I fine-tune one speaker at a time, multiple times, like upgrading weapons in a game?

**Comment by SWivid** [comment]
it's like:
you came up with many different weapons when start up, would probably try various builds or say playing styles
then if you wanna be a pro of a specific weapon, you make more practice on it when ranking but still do others at casual playing

samely, we generally start from a general base model, then finetune for specific-speaker(s) enhancement
if finetune each speaker at each time, would forget and omit some basic/cross ability; e.g. pace/strategy that may not be able learned with single case

> do I still need at least 10 hours of data per speaker

not sure but more balanced distribution is fine, e.g. 10 speakers each 1h? all up to experimental results

**Comment by wanfnc** [comment]
@SWivid Hi! I'm wondering what to do with the vocab. Should I copy and use the existing one, or should I create a new one using transcribed data?

**Comment by wanfnc** [comment]
Hi @SWivid,

"also remember to turn off use_ema when infer, at early stage of finetuning"

about this, I can't figure out how to turn off ema. Should I modify the code of "train.py" or "finetune_cli.py" ?


Secondly, auto-setting sets batch size 9,600 while my GPU VRAM is 10GB. Is it better to modify about 1600?

Always great thanks to you. And Happy New Year!

**Comment by SWivid** [comment]
https://github.com/SWivid/F5-TTS/blob/4872afef9f81117f4d4b373dc98ed5ed09d77039/src/f5_tts/infer/utils_infer.py#L230

it is to turn off here if when inference but not training or finetuning

> Secondly, auto-setting sets batch size 9,600 while my GPU VRAM is 10GB. Is it better to modify about 1600?

no big idea if oom-free, all fine thought

**Comment by yiwei0730** [comment]
@wanfnc 
Can I ask how you trained the Korean model? I found that the Korean and Japanese vocabs are missing in the original vocab.
And some of the original Korean vocabulary is pinyin and some is a complete word (there are more than a thousand words in Korean)
I would like to ask if you have done any processing before, and if there are any additional parts in vocab.txt that you can let me know. Thank you!

**Comment by wanfnc** [comment]
@yiwei0730 
Hi, I followed the training process as described in f5-tts. For the vocabulary, I manually created the metadata to supplement it. In other words, I used the same process but improved it through data preprocessing.
I recommend setting the Korean vocabulary to 'char' instead of 'pinyin'.
I would say the most important factors are the amount and quality of the dataset.

**Comment by yiwei0730** [comment]
wow if you used the char, that meaned you used thousands of the char in vocab?
what is your data duration you used for trained?

**Comment by wanfnc** [comment]
I don't remember. It was about 40 GB.
